File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        for(int j = nums.length - 1; j > 0; j--){
            for(int i = 0; i < j; i++){
                if(nums[i] + nums[j] == target){
                    return new int[]{i, j};
                }
            }
        }
        return new int[]{0};
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        
        #every subset with its sum hashmap

        res = []
        subset = []
        nums.sort()

        def dfs(i, t):
            if t == target:
                res.append(subset.copy())
                return
            if i == len(nums) or t > target:
                return
            
            #take it
            subset.append(nums[i])
            dfs(i, t + nums[i])

            #dont take it
            subset.pop()
            dfs(i + 1, t)
        
        dfs(0, 0)
        return res
